src/common/config/env.validation.ts

```diff
@@ -1,64 +1,72 @@
 import { plainToClass, Transform } from 'class-transformer';
 import {
-  IsEnum,
-  IsString,
-  IsOptional,
-  validateSync,
-  Min,
-  IsArray,
   ArrayMinSize,
-  IsInt,
+  IsArray,
   IsBoolean,
-  ValidateIf,
+  IsEnum,
+  IsInt,
   IsNotEmpty,
+  IsOptional,
   IsPositive,
+  IsString,
+  IsUrl,
+  Max,
+  Min,
+  ValidateIf,
+  validateSync,
 } from 'class-validator';
-import { Environment, LogLevel, LogFormat } from './interfaces';
+import { Environment, LogLevel, LogFormat, Network } from './interfaces';
 import { NonEmptyArray } from '@lido-nestjs/execution/dist/interfaces/non-empty-array';
 
 const toNumber =
   ({ defaultValue }) =>
   ({ value }) => {
-    if (value === '' || value == null || value == undefined) return defaultValue;
+    if (value === '' || value == null) return defaultValue;
     return Number(value);
   };
 
-export const toBoolean = (value: any): boolean => {
-  if (typeof value === 'boolean') {
-    return value;
-  }
+const toBoolean = ({ defaultValue }) => {
+  return function({ value }) {
+    if (value == null || value === '') {
+      return defaultValue;
+    }
 
-  if (typeof value === 'number') {
-    return !!value;
-  }
+    if (typeof value === 'boolean') {
+      return value;
+    }
 
-  if (!(typeof value === 'string')) {
-    return false;
-  }
+    const str = value.toString().toLowerCase().trim();
```

Comment:
what if it is number?

Reply:
I think that the previous code gave users too much unnecessary freedom in setting boolean values. I think it's better to be more strict here and allow users to set only "true" or "false" strings as valid values of boolean variables and throw errors otherwise.

But I understand that it is a breaking change. If we know or can reasonably assume that some KAPI users should be able to set "1" or "yes" as valid values of boolean variables for a good reason, or some users already doing it this way, we should revert back to the previous behavior.

What do you think?

Reply:
we have a lot of users, so i dont know 
i think in guides was true/false value, so they used it

Reply:
Based on the results of today's internal discussion, it has been decided to revert back support of numerical values and special constants like "yes" and "no", but return an error if the `undefined`, `null` or empty string was provided. I'll update the PR.
@Amuhar please, let me know if you have any objections.

Reply:
okay, but i agree that just boolean values look better